### PR TITLE
EVG-6986: Create granular global permissions

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: globals.go [hunk 1]
```diff
@@ -522,6 +528,14 @@ var (
 
 // permission levels
 var (
+	AdminSettingsEdit = PermissionLevel{
+		Description: "Edit admin settings",
+		Value:       10,
+	}
+	ProjectSettingsCreate = PermissionLevel{
```
Comment (john-m-liu) [inline globals.go]: This should be a separate permission, rather than a new level on the existing permission

File: rest/route/middleware.go [hunk 2]
```diff
@@ -442,6 +458,10 @@ func urlVarsToDistroScopes(r *http.Request) (string, int, error) {
 	return distroID, http.StatusOK, nil
 }
 
+func superUserResource(_ *http.Request) (string, int, error) {
```
Comment (john-m-liu) [inline rest/route/middleware.go]: This isn't needed if there's only 1 resource

File: globals.go [hunk 3]
```diff
@@ -566,6 +580,10 @@ var (
 		Description: "Not able to view logs",
 		Value:       0,
 	}
+	DistroSettingsCreate = PermissionLevel{
```
Comment (john-m-liu) [inline globals.go]: This should be a separate permission, rather than a new level on the existing permission

File: rest/route/middleware.go (hunk 2 above)
Comment (julianedwards) [reply rest/route/middleware.go]: right, but it needs to somehow get the name of that resource

**Comment by julianedwards** [comment]
ready for review

File: rest/route/middleware.go (hunk 2 above)
Comment (john-m-liu) [reply rest/route/middleware.go]: The name of the resource is passed in to the HasPermission function and the appropriate scope is looked up by gimlet

File: globals.go [hunk 4]
```diff
@@ -509,19 +510,34 @@ type PermissionLevel struct {
 }
 
 var (
+	// SuperUserPermissions resource ID.
+	SuperUserPermissionsID = "super_user"
+
+	// Admin permissions.
+	PermissionAdminSettings = "admin_settings"
 	// Project permissions.
+	PermissionProjectCreate    = "project_create"
```
Comment (john-m-liu) [inline globals.go]: The admin permissions should be grouped together to make it clear they aren't related to the project/distro permissions

**Pull request merged**